### PR TITLE
[RSDK-7692] Shutdown command

### PR DESCRIPTION
**Pull request by allisonschiang**
exit() and esp_restart() both return nothing, so I assume they cannot fail and did not return an error

### CODE REVIEW

File: micro-rdk/src/common/grpc.rs [hunk 1]
```diff
@@ -1212,6 +1213,16 @@ where
         self.encode_message(operation)
     }
 
+    // robot_shutdown will not return anything because will restart
+    fn robot_shutdown(&mut self, _: &[u8]) -> ! {
```
Comment (acmorrow) [inline micro-rdk/src/common/grpc.rs:1217]: You could just not take an argument here too. Nothing requires you to pass `payload` above, that I see.

File: micro-rdk/src/common/grpc.rs [hunk 2]
```diff
@@ -1212,6 +1213,16 @@ where
...
+        #[cfg(feature = "native")]
+        std::process::exit(0);
+        #[cfg(feature = "esp32")]
+        unsafe {
+            crate::esp32::esp_idf_svc::sys::esp_restart();
```
Comment (acmorrow) [inline micro-rdk/src/common/grpc.rs:1222]: Is it OK that `shutdown` for an esp32 is just going to reboot and restart the robot? What are the intended semantics of `shutdown`? Should this require some more active measures to get the robot running again, like powercycling the esp32? We could, for instance, enter deep sleep with no wake source instead.

**Review by acmorrow — APPROVED** [review]
The code LGTM with one small optional improvement. I'd like some clarification though on whether just restarting is an OK outcome on esp32 as opposed to taking some more drastic step. @stevebriskin and @npmenard I'd appreciate your thoughts. If `shutdown` is supposed to be estop-ish, what should happen here?

File: micro-rdk/src/common/grpc.rs (hunk 1 above)
Comment (allisonschiang) [reply micro-rdk/src/common/grpc.rs:1217]: I see some other requests that don't take any arguments still take a payload and don't use the value, like get_operations(), Resource_names(), etc... Is shutdown a different situation?

Comment (acmorrow) [reply micro-rdk/src/common/grpc.rs:1217]: I wonder if you get an unused warning above or something if you do that. OK, nevermind, just follow the pattern that is in use and go with what you have here now.

**Review by mattjperez — APPROVED** [review]
LGTM

**Pull request merged**